### PR TITLE
Add navsys support

### PR DESCRIPTION
**Pull request by CramBL**
#### PR Type

- [x] 💰 Feature
- [ ] 🪲 Bug Fix
- [x] 👷‍♀️ Refactor
- [ ] 📖 Documentation Update

#### Checklist

- [x] 🌞 Changelog updated

### CODE REVIEW

File: crates/skytem_logs/src/navsys/entries/he.rs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+}
+
+impl AltimeterEntry {
+    /// This value is placeholder for an invalid reading
+    pub(crate) const INVALID_VALUE: &str = "99999.99";
+
+    pub(crate) fn timestamp_ns(&self) -> f64 {
+        self.timestamp
+            .timestamp_nanos_opt()
+            .expect("timestamp as nanoseconds out of range") as f64
```
Comment (avi-skytem) [inline crates/skytem_logs/src/navsys/entries/he.rs:23]: Do you want the code to panic here if timestamp is out of range?

**Review by avi-skytem — APPROVED** [review]
I reviewed the code, but I didn't go through each entry file in detail. However, some of the comments I left may apply across multiple files.

I’ve approved the changes as this review was also a learning experience for me. I don't expect you to address all of the comments.

File: crates/log_if/src/parseable.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    /// Returns whether or not the buffer is a valid instance of [`Self`]
+    ///
+    /// Implementers should read and verify the buffer until there's no doubt that this is
+    /// a buffer of [`Self`] content, e.g. an `Mbed PID Log`. If several supported formats
+    /// returns true for a given buffer, there's no way to know how to parse it.
+    ///
+    /// Performance is not a concern here as we are only doing this check once per file
+    /// and only when the file is loaded, it is much more important that the check is
+    /// extremely rigorous.
+    fn is_buf_valid(buf: &[u8]) -> bool;
```
[the remark this answers is not in the document]
Comment (CramBL) [reply crates/log_if/src/parseable.rs:38]: The check cannot be optional considering this is how we determine what kind of format the input buffer represents

File: crates/skytem_logs/src/generator.rs [hunk 3]
```diff
@@ -34,15 +34,30 @@ impl GeneratorLog {
...
-        let mut buf_reader = BufReader::new(reader);
+    fn entries(&self) -> &[Self::Entry] {
+        &self.entries
+    }
+}
+
+impl Parseable for GeneratorLog {
+    const DESCRIPTIVE_NAME: &str = "Legacy Generator Log";
+    fn is_buf_valid(buf: &[u8]) -> bool {
+        let mut bufreader = BufReader::new(buf);
```
[the remark this answers is not in the document]
Comment (CramBL) [reply crates/skytem_logs/src/generator.rs:45]: doesn't matter much imo

File: crates/skytem_logs/src/navsys/entries.rs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                format!(
+                "Expected NavSysSps entry line but line is too short to be a NavSysSps entry. Line length={}, content={}", line.len(), line
+            )));
+        }
+        let first_three_chars = &line[..3];
+        let entry: Self = match first_three_chars {
+            "HE1" | "HE2" => {
+                let he = AltimeterEntry::from_str(&line)
+                    .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
+                debug_assert!(he.id == 1 || he.id == 2);
```
[the remark this answers is not in the document]
Comment (CramBL) [reply crates/skytem_logs/src/navsys/entries.rs:53]: `debug_assert` is only active when it is build in debug mode, `assert` would be the production equivelant

File: crates/skytem_logs/src/navsys/entries/he.rs (hunk 1 above)
Comment (CramBL) [reply crates/skytem_logs/src/navsys/entries/he.rs:23]: YES!!! The entire program would be invalid if the timestamp is too large to be represented in nanoseconds

File: crates/skytem_logs/src/navsys/entries.rs [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        let bytes_read = reader.read_line(&mut line)?;
+        // just a sanity check, it is definitely invalid if it is less than 10 characters
+        if line.len() < 10 {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidData,
+                format!(
+                "Expected NavSysSps entry line but line is too short to be a NavSysSps entry. Line length={}, content={}", line.len(), line
+            )));
+        }
+        let first_three_chars = &line[..3];
```
[the remark this answers is not in the document]
Comment (CramBL) [reply crates/skytem_logs/src/navsys/entries.rs:48]: ...There's a check for `line.len() < 10` on line 41.

`line.starts_with("...")` might be prettier if there was only one case, but there's 7 cases and I want to specifically check against the first 3 letters. The match case is quite concise as is, but there might be a better way to do it. Not with `line.starts_with("...")` though, I assume. But if you can suggest something then I'm all ears.

File: crates/skytem_logs/src/navsys/entries/he.rs [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        let timestamp = DateTime::from_naive_utc_and_offset(naive_dt, Utc);
+
+        // Parse altitude
+        let altitude = if parts[8] == Self::INVALID_VALUE {
+            None
+        } else {
+            Some(
+                parts[8]
+                    .parse()
+                    .map_err(|e: ParseFloatError| AltimeterParseError::Altitude(e.to_string()))?,
```
[the remark this answers is not in the document]
Comment (CramBL) [reply crates/skytem_logs/src/navsys/entries/he.rs:79]: I agree in the general case, but all these parse errors are already way too much error handling, this stuff will never happen so I'm inclined to not spend more effort on it

**Pull request merged**